jdbc/basics/src/main/java/example/springdata/jdbc/basics/aggregate/AggregateContext.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.data.jdbc.core.DataAccessStrategy;
+import org.springframework.data.jdbc.core.DefaultDataAccessStrategy;
+import org.springframework.data.jdbc.core.DelegatingDataAccessStrategy;
+import org.springframework.data.jdbc.core.SqlGeneratorSource;
+import org.springframework.data.jdbc.mapping.event.BeforeSave;
+import org.springframework.data.jdbc.mapping.model.*;
+import org.springframework.data.jdbc.repository.config.EnableJdbcRepositories;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.lang.Nullable;
+
+import javax.sql.DataSource;
+import java.sql.Clob;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author Jens Schauder
+ */
+@Configuration
+@EnableJdbcRepositories
+public class AggregateContext {
+
+	@Bean
+	public ApplicationListener<?> idSetting() {
+
+		final AtomicInteger id = new AtomicInteger(0);
+
+		return (ApplicationListener<BeforeSave>) event -> {
+
+			Object entity = event.getEntity();
+			if (entity instanceof LegoSet) {
+
+				LegoSet legoSet = (LegoSet) entity;
+				if (legoSet.getId() == 0) {
+					legoSet.setId(id.incrementAndGet());
+				}
+
+				Manual manual = legoSet.getManual();
+				if (manual != null) {
+					manual.setId((long) legoSet.getId());
+				}
+			}
+		};
+	}
+
+	@Bean
+	public NamingStrategy namingStrategy() {
+
+		Map<String, String> tableAliases = new HashMap<String, String>();
+		tableAliases.put("Manual", "Handbuch");
+
+		Map<String, String> columnAliases = new HashMap<String, String>();
+		columnAliases.put("LegoSet.intMaximumAge", "maxAge");
+		columnAliases.put("LegoSet.intMinimumAge", "minAge");
+		columnAliases.put("Handbuch.id", "Handbuch_id");
+
+		Map<String, String> reverseColumnAliases = new HashMap<String, String>();
+		reverseColumnAliases.put("manual", "Handbuch_id");
+
+		Map<String, String> keyColumnAliases = new HashMap<String, String>();
+		keyColumnAliases.put("models", "name");
+
+		return new DefaultNamingStrategy() {
+
+			@Override
+			public String getColumnName(JdbcPersistentProperty property) {
+
+				String defaultName = super.getColumnName(property);
+				String key = getTableName(property.getOwner().getType()) + "." + defaultName;
+				return columnAliases.getOrDefault(key, defaultName);
+			}
+
+			@Override
+			public String getTableName(Class<?> type) {
+
+				return tableAliases.getOrDefault(super.getTableName(type), super.getTableName(type));
+			}
+
+			@Override
+			public String getReverseColumnName(JdbcPersistentProperty property) {
+				return reverseColumnAliases.getOrDefault(property.getName(), super.getReverseColumnName(property));
```

Comment:
Should we switch to `….computeIfAbsent(…)` here to avoid the call to `super.…` in case a value has been registered already and avoid duplicate lookups?